src/main/java/fact/utils/SamplePedestalEvent.java

```diff
@@ -0,0 +1,385 @@
+/**
+ *
+ */
+package fact.utils;
+
+import fact.Utils;
+import fact.container.PreviousEventInfoContainer;
+import fact.io.hdureader.FITSStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import stream.Data;
+import stream.ProcessContext;
+import stream.StatefulProcessor;
+import stream.annotations.Parameter;
+import stream.expressions.version2.Condition;
+import stream.expressions.version2.ConditionFactory;
+import stream.io.JSONStream;
+import stream.io.SourceURL;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import static java.lang.Math.ceil;
+
+/**
+ * This processors loads the given noise database and randomly chooses for each event processed a noise event,
+ * which is then loaded into the item, with the keys prepended with the string prependKey.
+ *
+ * @author mbulinski
+ *
+ */
+public class SamplePedestalEvent implements StatefulProcessor {
```

Comment:
We made all parameters public in the other processors.

Reply:
Fixed.

Reply:
The fields are all stil private